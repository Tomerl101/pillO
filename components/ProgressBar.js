import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Bar } from 'react-native-progress';
import { Text } from './Texts/Text';

export const ProgressBar = () => {
  return (
    <View>
      <View style={styles.barTextContainerStyle}>
        <Text style={styles.bigFontStyle}>03</Text>
        <Text style={styles.greenColor}>/ 10 dose taken</Text>
      </View>
      <View style={styles.leftMargin}>
        <Bar
          progress={0.3}
          color="white"
          borderWidth={0}
          unfilledColor="#7bedca"
          height={12}
          width={300}
          borderRadius={100}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  barTextContainerStyle: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    marginLeft: 38,
    marginTop: 8,
    marginBottom: 12
  },
  bigFontStyle: {
    fontSize: 28
  },
  greenColor: {
    color: '#1E6B53'
  },
  leftMargin: {
    marginLeft: 42
  }
});
