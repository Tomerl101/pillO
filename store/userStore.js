import { observable, action, computed, toJS } from 'mobx';
import perscriptionsMockData from '../mock/medicine_perscription.json';

class UserStore {
  @observable currentUser = {};
  @observable isLoading = false;
  @observable perscriptionsList = [];

  constructor() {
    this.loadPerscriptions();
  }

  @computed get takenPerscriptionsCount() {
    return 2;
    // return this.perscriptionsList.filter(todo => todo.taken === true).length;
  }

  @computed get totalPerscriptionCount() {
    return this.perscriptionsList.length;
  }

  @action loadPerscriptions() {
    this.isLoading = true;
    this.perscriptionsList = perscriptionsMockData;
    this.isLoading = false;
  }
}
export default new UserStore();
