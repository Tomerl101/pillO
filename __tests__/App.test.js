import 'react-native'
import React from 'react'
import '../setupTest'
import userStore from '../store/userStore'
import ProgressBar from '../components/ProgressBar'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'

describe('App snapshot', () => {
  test('hydrate favorite images to state from async storage', function() {
    expect([]).toHaveLength(0)
  })

  test('should return taken perscription', function() {
    userStore.setFilter('TAKEN')
    expect(userStore.filteredPerscription).toHaveLength(1)
  })

  test('should load perscription', function() {
    expect(userStore.perscriptionsList).toHaveLength(4)
  })

  test('should set taken perscription', () => {
    userStore.setTaken(2)
    expect(userStore.takenPerscriptionsCount).toBe(2)
  })

  test('renderer ProgressBar with 2 taken perscription', () => {
    const taken = userStore.takenPerscriptionsCount
    const total = userStore.totalPerscriptionCount

    const tree = renderer.create(<ProgressBar taken={taken} total={total} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
