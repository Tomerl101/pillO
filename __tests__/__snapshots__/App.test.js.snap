// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renderer ProgressBar with 2 taken perscription 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 38,
        "marginTop": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Raleway-Regular",
            "fontSize": 16,
          },
          Object {
            "fontSize": 28,
          },
        ]
      }
    >
      2
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Raleway-Regular",
            "fontSize": 16,
          },
          Object {
            "color": "#1E6B53",
          },
        ]
      }
    >
      / 4 dose taken
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 42,
      }
    }
  >
    <View
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={false}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0.5}
      style={
        Array [
          Object {
            "backgroundColor": "#7bedca",
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 0,
            "overflow": "hidden",
            "width": 300,
          },
          undefined,
        ]
      }
      useNativeDriver={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 12,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateX": -75,
              },
              Object {
                "scaleX": 0.50005,
              },
            ],
          }
        }
      />
    </View>
  </View>
</View>
`;
